jerry-ext/include/jerryscript-ext/module.h

```diff
@@ -22,6 +22,24 @@
 #define JERRYX_NATIVE_MODULES_SUPPORTED
 #endif /* __GNUC__ */
 
+/**
+ * Declare the structure for module resolvers.
+ */
+typedef struct
+{
+  jerry_value_t (*get_canonical_name) (const jerry_value_t name); /**< module name to canonicalize */
+  bool (*resolve) (const jerry_value_t canonical_name, /**< requested module's canonical name */
+                   jerry_value_t *result); /**< resulting module */
```

Comment:
What's your rationale for passing around the name strings as `jerry_value_t`s?
I think most module resolvers will convert it right back to a C string.
Is it to avoid having to deal with memory management of the C strings?

Reply:
Yeah, and I was actually thinking that, for path resolution, you might not even convert the whole value to a C string, but you would rather use fixed size chunks to gradually retrieve the string and maybe assemble the canonical name in a similarly gradual fashion. Then, unless you had a directory with a really, really, really long name, you wouldn't really be allocating all that much.

Reply:
Call it streamed path resolution.

Reply:
Another very important reason: the module cache is stored as a JavaScript object, so I have to convert a C string back to a `jerry_value_t` in order to check if that JavaScript object has a module at the module name's key.

Reply:
Ah I see, yeah that makes sense.